File: 0611-Valid Triangle Number/0611-Valid Triangle Number.cpp
class Solution {
public:
    int triangleNumber(vector<int>& nums) {
        int n = nums.size();
        if (n < 3)
        {
            return 0;
        }
        
        sort(nums.begin(), nums.end());
        int count = 0;
        for (int i = n - 1; i >= 2; --i)
        {
            int c = nums[i];
            int left = 0, right = i - 1;
            while (left < right)
            {
                if (nums[left] + nums[right] <= c)
                {
                    ++left;
                }
                else
                {
                    count += right - left;
                    --right;
                }
            }
        }
        
        return count;
    }
};

File: 0601-0700/0611-Valid Triangle Number/0611-Valid Triangle Number.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        nums.sort()
        count = 0
        for i in range(2, len(nums)):
            left = 0
            right = i - 1
            while left < right:
                if nums[left] + nums[right] <= nums[i]:
                    left += 1
                else:
                    count += right - left
                    right -= 1
        return count
